CONTRIBUTING.md

```diff
@@ -0,0 +1,70 @@
+# Join the development
+* Before you join development, please set up the project on your local machine, run it and go through the application completely. Press on any button you can find and see where it leads to. Explore.You'll be more familiar with what is where and might even get some cool ideas on how to improve various aspects of the app.
+* If you would like to work on an issue, drop in a comment at the issue. If it is already assigned to someone, but there is no sign of any work being done, please free to drop in a comment so that the issue can be assigned to you if the previous assignee has dropped it entirely.
+
+# Contributing
+
+When contributing to this repository, please first discuss the change you wish to make via issue, or the official [slack channel](https://slack-ssl-openfoodfacts.herokuapp.com/).
+
+## Pull Request Process
+
+1. Ensure any install or build dependencies are removed before the end of the layer when doing a 
+   build.
+2. Check that there are no conflicts and your request passes [Travis](Travis-ci.org) build. Check the log of the pass test if it fails the build.
+3. Give the description of the issue that you want to resolve in the pull request message. The format of the commit message to be fixed    - **Fixes #[issue number] [Description of the issue]** Example: **Fixes #529: Add toast warning in `MainActivity.java`** 
+4. Wait for the maintainers to review your pull request and do the changes if requested.
+
+## Contributions Best Practices
+
+### Commits
+
+* Write clear meaningful git commit messages (Do read [here](http://chris.beams.io/posts/git-commit/))
+* Make sure your PR's description contains GitHub's special keyword references that automatically close the related issue when the PR is merged. (For more info click [here]( https://github.com/blog/1506-closing-issues-via-pull-requests))
+* When you make very very minor changes to a PR of yours (like for example fixing a failing Travis build or some small style corrections or minor changes requested by reviewers) make sure you squash your commits afterward so that you don't have an absurd number of commits for a very small fix. (Learn how to squash at [here](https://davidwalsh.name/squash-commits-git )
+* When you're submitting a PR for a UI-related issue, it would be really awesome if you add a screenshot of your change or a link to a deployment where it can be tested out along with your PR. It makes it very easy for the reviewers and you'll also get reviews quicker.
+
+### Feature Requests and Bug Reports
+* When you file a feature request or when you are submitting a bug report to the [issue tracker](https://github.com/openfoodfacts/openfoodfacts-server/issues), make sure you add steps to reproduce it. Especially if that bug is some weird/rare one.
+
+## Code of Conduct
```

Comment:
There's a CoC in the Wiki: https://en.wiki.openfoodfacts.org/Code_of_conduct Maybe we should link to that instead?

Reply:
Okay thats great I will link it to that.